lib/Modules/ComputeHooks/ComputeHooks.psm1

```diff
@@ -671,20 +689,41 @@ function Start-UpgradeOpenStackVersion {
 
 function Set-HyperVUniqueMACAddressesPool {
     $registryNamespace = "HKLM:\Software\Microsoft\Windows NT\CurrentVersion\Virtualization"
-
     $randomBytes = @(
         [byte](Get-Random -Minimum 0 -Maximum 255),
         [byte](Get-Random -Minimum 0 -Maximum 255)
     )
-
     # Generate unique pool of MAC addresses
     $minMacAddress = @(0x00, 0x15, 0x5D, $randomBytes[0], $randomBytes[1], 0x00)
     Set-ItemProperty -Path $registryNamespace -Name "MinimumMacAddress" -Value ([byte[]]$minMacAddress)
-
     $maxMacAddress = @(0x00, 0x15, 0x5D, $randomBytes[0], $randomBytes[1], 0xff)
     Set-ItemProperty -Path $registryNamespace -Name "MaximumMacAddress" -Value ([byte[]]$maxMacAddress)
 }
 
+function Set-S2DHealthChecksRelation {
+    $s2dRelationCreated = Confirm-JujuRelationCreated -Relation 's2d'
+    $csvRelationCreated = Confirm-JujuRelationCreated -Relation 'csv'
+    if(!$s2dRelationCreated -or !$csvRelationCreated) {
+        return
+    }
+    # In case cluster nodes go down, the Windows fail-over cluster will set the CSV's to offline
+    # and they need to be manually brought back online when nodes come up again. We need to trigger
+    # health checks for the S2D cluster whenever nodes boot for the first time. The s2d-proxy
+    # charm will make sure that the CSV for this unit is online and mounted.
+    $systemStartups = Get-WinEvent -FilterHashtable @{
+        'LogName'='Security'
+        'ID' = 4608 # This event is raised only when Windows is starting up.
+    } -ErrorAction SilentlyContinue
+    $rids = Get-JujuRelationIds 's2d-health-check'
+    foreach($rid in $rids) {
+        Set-JujuRelation -RelationId $rid -Settings @{
+            'system-startups' = $systemStartups.Count
```

Comment:
Yep. During my tests, I saw that some of these events get indeed deleted for a long running system. (probably event viewer keep a number of maximum logs).

But my idea was to ping the `s2d-proxy` charm whenever the node comes up, so that `s2d-proxy` charm can check the health of all the `csv`s or any other potential s2d cluster health operations.